### DISCUSSION TITLE
how do I add hex color on colorpicker?

### DISCUSSION THREAD

**Discussion by helloeyesimple** [awcodes/filament-tiptap-editor#244] [Q&A]
I'm using default mode, and know that I can change font color using colorpicker. But I want to use specific hex instead of picking it manually. how can I achieve that?

**Comment by awcodes** [comment]
I'll make a note to see if i can figure something out. It's just not that simple since it's a form element that needs to stay isolated from the normal form's state. But at the moment you can't do anything from your side to make it work.

I'm open to PRs if you can solve it in the mean time thought.

Thanks.

**Comment by 000yuying** [comment]
Here's what I currently have that's similar. I hope it can help you.
 
(your-project/resources/js/tiptap/extensions.js)
1.Following the recommendations in the document, add a new file called extensions.js.
ex:
import Color from "@tiptap/extension-color";
window.TiptapEditorExtensions = {
    color: [Color.configure({
        types: ['textStyle'],
    })],
};

(your-project/resources/views/components/tools/color.blade.php)
2.Add new  blade.php
<x-filament-tiptap-editor::dropdown-button
    label="{{ trans('filament-tiptap-editor::editor.color.label') }}"
    active="color"
    icon="color"
    :list="false"
>
    <div
        x-data="{
            state: editor().getAttributes('textStyle').color || '#000000',

            init: function () {
                if (!(this.state === null || this.state === '')) {
                    this.setState(this.state)
                }
            },

            setState: function (value) {
                this.state = value
            }
        }"
        x-on:keydown.esc="isOpen() && $event.stopPropagation()"
        class="relative flex-1 p-1"
    >
        <input
            type="text"
            x-model="state"
            class="w-full p-2 border border-gray-300 rounded-md"
        />

        <tiptap-hex-color-picker x-bind:color="state" x-on:color-changed="setState($event.detail.value)"></tiptap-hex-color-picker>

        <div class="w-full flex gap-2 mt-2">
            <x-filament::button
                x-on:click="editor().chain().focus().setColor(state).run(); $dispatch('close-panel')"
                size="sm"
                class="flex-1"
            >
                {{ trans('filament-tiptap-editor::editor.color.choose') }}
            </x-filament::button>

            <x-filament::button
                x-on:click="editor().chain().focus().unsetColor().run(); $dispatch('close-panel')"
                size="sm"
                class="flex-1"
                color="danger"
            >
                {{ trans('filament-tiptap-editor::editor.color.remove') }}
            </x-filament::button>
        </div>
    </div>
</x-filament-tiptap-editor::dropdown-button>


(your-project/app/TiptapExtensions/color.php)
3.Add new  Extensions.php
<?php

namespace App\TiptapExtensions;

class color extends \Tiptap\Nodes\Heading
{
    public static $name = 'color';
}


4.Then, in config add custom extensions
ex:
'extensions' => [
        [
            'id' => 'color',
            'name' => 'color',
            'button' => 'tools.color',
            'parser' => \App\TiptapExtensions\color::class,
        ]
 ]
 
 
and npm run build 
you will have this


<img width="814" alt="image" src="https://github.com/awcodes/filament-tiptap-editor/assets/109067769/310005e0-39fa-4bac-b6cd-ed8b9b034507">
